modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/LZ4CompressionTestCase.java

```diff
@@ -0,0 +1,186 @@
+     */
+    @BeforeClass(alwaysRun = true)
+    public void init() throws XPathExpressionException {
+        super.init(TestUserMode.SUPER_TENANT_USER);
+    }
+
+    /**
+     * Set allowCompression to true so that broker will compress messages before storing into the database, and
+     * reduce maximum content chunk size to 100 from default value (65500); to create more content chunks
+     * from compressed content, to check chunk data retrieval.
+     *
+     * @throws XPathExpressionException
+     * @throws java.io.IOException
+     * @throws org.apache.commons.configuration.ConfigurationException
+     * @throws org.wso2.carbon.integration.common.utils.exceptions.AutomationUtilException
+     */
+    @BeforeClass
+    public void setupConfiguration() throws XPathExpressionException, IOException, ConfigurationException,
+            AutomationUtilException {
+
+        super.serverManager = new ServerConfigurationManager(automationContext);
+        String defaultMBConfigurationPath = ServerConfigurationManager.getCarbonHome() +
+                File.separator + "repository" + File.separator + "conf" + File.separator + "broker.xml";
+
+        ConfigurationEditor configurationEditor = new ConfigurationEditor(defaultMBConfigurationPath);
+
+        configurationEditor.updateProperty(AndesConfiguration.PERFORMANCE_TUNING_ALLOW_COMPRESSION, "true");
+        configurationEditor.updateProperty(AndesConfiguration.PERFORMANCE_TUNING_MAX_CONTENT_CHUNK_SIZE, "100");
+
+        configurationEditor.applyUpdatedConfigurationAndRestartServer(serverManager);
+    }
+
+    /**
+     * Test the message content integrity of a single message when compression is enabled, by comparing the sent and
+     * received message content which spreads over several message content chunks.
+     *
+     * @throws AndesClientConfigurationException
+     * @throws IOException
+     * @throws JMSException
+     * @throws NamingException
+     * @throws AndesClientException
+     */
+    @Test(groups = "wso2.mb", description = "Compressed message content validation test case")
+    public void performQueueContentSendCompressReceiveTestCase() throws AndesClientConfigurationException, IOException,
+            JMSException, NamingException, AndesClientException, XPathExpressionException {
+
+        // Reading message content
+        char[] inputContent = new char[SIZE_TO_READ];
+        try {
+            BufferedReader inputFileReader = new BufferedReader(
+                    new FileReader(AndesClientConstants.MESSAGE_CONTENT_INPUT_FILE_PATH_WITHOUT_REPETITIONS_256KB));
+            inputFileReader.read(inputContent);
+        } catch (FileNotFoundException e) {
+            log.warn("Error locating input content from file : "
+                    + AndesClientConstants.MESSAGE_CONTENT_INPUT_FILE_PATH_WITHOUT_REPETITIONS_256KB);
+        } catch (IOException e) {
+            log.warn("Error reading input content from file : "
```

Comment:
Better, not catch these exceptions and throw them, test will be recorded as failed and the exception will be given as the reason for the test failure in the testng report.